src/svc_auth_gss.c

```diff
@@ -676,6 +676,8 @@ svcauth_gss_destroy(SVCAUTH *auth)
 		gss_release_buffer(&min_stat, &gd->pac.ms_pac);
 
 	gss_release_buffer(&min_stat, &gd->checksum);
+
+	mutex_unlock(&gd->lock);
```

Comment:
It's actually safe to destroy a locked mutex, if you own the lock.  Not a big deal, though.